pkg/alertmanager/amcfg.go

```diff
@@ -1299,6 +1306,51 @@ func makeNamespacedString(in string, crKey types.NamespacedName) string {
 	return crKey.Namespace + "/" + crKey.Name + "/" + in
 }
 
+func (cb *configBuilder) convertSMTPConfig(ctx context.Context, out *globalConfig, in monitoringv1.GlobalSMTPConfig, crKey types.NamespacedName) error {
+	if out == nil {
+		return errors.New("globalConfig is required")
+	}
```

Comment:
1 line before calling this function we're declaring `out := &globalConfig{}`, therefore `out` will never be nil. There's no need for this if :)

Reply:
Yes, you are right. I thought at first that `convertSMTPConfig` should not be aware of it. But maybe since they are both methods of cb it is ok to remove this check?